nkms/crypto/signature.py

```diff
@@ -1,57 +1,59 @@
 from nkms.crypto import api as API
-from nkms.keystore.keypairs import PublicKey
+from umbral.keys import UmbralPublicKey
+from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature, encode_dss_signature
 
 
-class Signature(bytes):
+class Signature(object):
     """
     The Signature object allows signatures to be made and verified.
     """
-    _EXPECTED_LENGTH = 65
+    _EXPECTED_LENGTH = 64  # With secp256k1 and BLAKE2b(64).
```

Comment:
The hash function doesn't affect this, only the order of the curve

Reply:
The digest length of the hash affects it, no?  Or do I have that wrong?

Reply:
No, both r and s are integers modulo the order of the curve group. The hash is irrelevant to that matter.
BTW, the choice of a 512-bits hash is somewhat weird since the curve order is 256 bits. @tuxxy do you know what's the performance of blake2b with 32 bytes vs 64 bytes?

Reply:
Oh right - that makes sense.  

As for the digest: it's sadly a moot(-ish) point, as cryptography.io only allows `sign()` with 64 bytes.

Reply:
https://cryptography.io/en/latest/hazmat/primitives/asymmetric/ec/#cryptography.hazmat.primitives.asymmetric.ec.ECDSA

The example in here uses a 32-bytes hash function

Reply:
Perhaps there's something (in the internals of cryptography.io) that is specific to BLAKE2b - trying to use a 32-bit digest raises an exception.

Reply:
https://cryptography.io/en/latest/hazmat/primitives/cryptographic-hashes/#cryptography.hazmat.primitives.hashes.BLAKE2b

You're right, man. This is a cryptography.io issue. Nevermind then. 

The thing is that the message is first hashed to 64 bytes, but afterwards it's reduced to the order of the curve group (32 bytes), which makes it somewhat pointless to hash to 64 bytes initially. In any case this doesn't affect security.

Reply:
Sure, but it might affect performance, especially for a high-throughput Alice or Bob.